CMakeLists.txt

```diff
@@ -51,7 +51,7 @@ set(CMAKE_CXX_FLAGS
 	-Werror=uninitialized
 	-Werror=init-self
 	-Wcast-qual
-	-Wconversion
+	-Wno-conversion
```

Comment:
Why did you turn of conversion warning, this is important for the embedded sign to force user to realize if he is wasting casts.